spooky-doodle/Bangazon-Platform-API
# Order

# Description

- Created an Order model to represent the corresponding table in the database.
- Created an Order controller API to handle GET,POST,PUT, and DELETE requests
- Created n Order class in the testing solution to test each API call
- Conditionally embeds Orders based on queries 


Fixes # (issue)

## Type of change

Please delete options that are not relevant.


# Testing Instructions

To run tests,

- Pull down the branch named product-type from GitHub
- Open the Test Explorer in Visual Studio and run all tests

Tests

1. Checks to make sure the GET ALL request returns a 200 status code and one or more product types from the database
2. Checks to make sure the GET All request returns a 200 status code and one or more products embedded in the order
3. Checks to make sure the GET All request returns a 200 status code and one or customer is embedded in the order
4 Checks to make sure the GET ALL request returns a 200 status code and returns only completed orders
5. Checks to make sure the GET ALL request returns a 200 status code and returns only uncompleted orders
6. Checks to make sure the GET ONE request returns a 200 status code, the item coming back matches the the first item in the database, and the item returned in not null
7. Checks to make sure the GET ONE request returns a 200 status code, the item coming back matches the the first item in the database, and the item returned in not null, and has products embedded
8. Checks to make sure the GET ONE request returns a 200 status code, the item coming back matches the the first item in the database, and the item returned in not null, and has a customer embedded
9. Checks that the POST request returns a 201 status code and the objects name that is returned is the same as the object passed in the test
Checks that the DELETE request returns a 204 status code and deletes the object that was made during the POST test
10. Checks that the PUT request returns a 204 status code
Preforms a GET on the updated object to check for a 200 status code and that the updated objects name is the database is the same as the name that you created in the test



# Checklist:
